### DISCUSSION TITLE
Using ffcv for self-supervised learning

### DISCUSSION THREAD

**Discussion by chengxuz** [libffcv/ffcv#76] [Q&A] [answered]
Thanks for this amazing work!

I am working on self-supervised learning algorithms, mainly contrastive learning algorithms, and wondering how this work can be used to accelerate the training there. The key differences in the data loader part are that each image yields two or even more independent data augmentations (sometimes these augmentations even require different pipelines) and also data augmentations typically include color jittering and gaussian blurring. Do you have any idea on how these two things can be implemented in ffcv?

**Comment by andrewilyas** [comment]
Hi @chengxuz ! We are currently thinking of the right way to support putting the same field through multiple pipelines in FFCV. As an easy workaround in the meantime, I think you can use the suggestion from this issue: https://github.com/libffcv/ffcv/issues/77#issuecomment-1018900926 . Basically, just create two Loaders, give them the same batch ordering seed, and then zip them together.

**Comment by GuillaumeLeclerc** [comment, marked as answer]
Hello, this is planned under #82. Feel free to track this issue
